### ISSUE TITLE
`sc.external.pp.magic`: what is `n_jobs`'s default _really_?

### ISSUE THREAD

**Issue by alexlenail** [scverse/scanpy#2387]
[scanpy docs](https://scanpy.readthedocs.io/en/stable/generated/scanpy.external.pp.magic.html): 

> n_jobs : [Optional](https://docs.python.org/3/library/typing.html#typing.Optional)[[int](https://docs.python.org/3/library/functions.html#int)] (default: None)
Number of threads to use in training. All cores are used by default.

[magic docs](https://magic.readthedocs.io/en/stable/api.html):

> n_jobs (integer, optional, default: 1) – The number of jobs to use for the computation. If -1 all CPUs are used. If 1 is given, no parallel computing code is used at all, which is useful for debugging.

scanpy code: 

https://github.com/scverse/scanpy/blob/536ed15bc73ab5d1131c0d530dd9d4f2dc9aee36/scanpy/external/pp/_magic.py#L164

https://github.com/scverse/scanpy/blob/536ed15bc73ab5d1131c0d530dd9d4f2dc9aee36/scanpy/_settings.py#L82

I'm guessing the scanpy docs are wrong when they say "All cores are used by default." ?

**Comment by jpmcga** [comment]
^^I've empirically confirmed this; been seeing multi-fold speedups when I set `n_jobs=-1` corresponding to the number of cores, versus running with the default.  From what I can tell, running without specifying `n_jobs` results in 1 core being used.

**Comment by flying-sheep** [comment]
Yeah, we definitely need to improve this. #2390 is also relevant here
